lib/logging.dart

```diff
@@ -364,3 +366,187 @@ class LogRecord {
+    name: record.loggerName,
+    time: record.time,
+    error: record.error,
+    level: record.level.value,
+    stackTrace: record.stackTrace,
+  );
+});
+
+/// Log Managers record messages only written to loggers explicitly enabled by
+/// name.
+///
+/// For example, in the given source, only messages to `loggerA` will be sent to
+/// the `onRecord` stream.
+///
+/// ```
+/// var loggerA = Logger('loggerA');
+/// var loggerA = Logger('loggerB');
+///
+/// var manager = LogManager()..enableLogging('loggerA');
+///
+/// loggerA.info('hello from logger A'); // recorded
+/// loggerB.info('here goes nothing');   // ignored
+/// ```
+///
+/// Log managers can be created but the provided [logManager] should suffice
+/// for most purposes.  The shared manager redirects recorded messages to the
+/// developer log.
+///
+/// **Important note:** managed loggers respect logger `root` level configuration.
+/// For example, in the following, `logger` only reports `severe` messages
+/// as expected.
+///
+/// ```
+/// Logger.root.level = Level.SEVERE;
+///
+/// ...
+///
+/// var logger = Logger('my.logger');
+/// logManager.enable('my.logger');
+/// try {
+///   logger.info('doing something'); // ignored
+///   ...
+/// } catch(e, stackTrace) {
+///   logger.severe('uh-oh!, e, stackTrace); // recorded
+/// }
+/// ```
+///
+/// * To create loggers, see [Logger].
+/// * To enable or disable a logger, use [enableLogging].
+/// * To query channel enablement, use [shouldLog].
+class LogManager {
+  static bool _initialized;
+
+  final Set<String> _enabledLoggers = Set<String>();
+
+  final StreamController<LogRecord> _logController =
+      StreamController.broadcast(sync: true);
+  final StreamController<String> _loggerAddedBroadcaster =
+      StreamController.broadcast();
+  final StreamController<String> _loggerEnabledBroadcaster =
+      StreamController.broadcast();
+
+  LogManager() {
+    Logger.root.onRecord.listen((record) {
+      if (shouldLog(record.loggerName)) {
+        _logController.add(record);
+      }
+    });
+  }
+
+  /// Returns a stream of messages added to loggers enabled by this manager.
+  Stream<LogRecord> get onRecord => _logController.stream;
+
+  /// Enable (or disable) logging of messages sent to the given [logger].
+  void enableLogging(String logger, {bool enable = true}) {
+    enable ? _enabledLoggers.add(logger) : _enabledLoggers.remove(logger);
+    _loggerEnabledBroadcaster.add(logger);
+  }
+
+  /// Whether messages to the given [logger] should be logged by this manager.
+  /// @see [enableLogging]
+  bool shouldLog(String logger) => _enabledLoggers.contains(logger);
+
+  /// Called to register service extensions.
+  void initServiceExtensions() {
+    // Avoid double initialization.
+    if (_initialized == true) {
+      return;
+    }
+
+    // Fire events for new loggers.
+    _loggerAddedBroadcaster.stream.listen((String name) {
+      developer.postEvent('logging.logger.added', <String, dynamic>{
```

Comment:
logging.logger seems a little redundant.
maybe
ext.logger.added
to match the
ext.flutter convention?